service/rpc/share.go

```diff
@@ -0,0 +1,77 @@
+package rpc
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+
+	"github.com/celestiaorg/celestia-node/service/header"
+)
+
+const namespacedSharesEndpoint = "/namespaced_shares"
+
+var nIDKey = "nid"
+
+func (h *Handler) handleSharesByNamespaceRequest(w http.ResponseWriter, r *http.Request) {
```

Comment:
Shouldn't the response be a struct that isn't just the raw shares but instead also for which height we are returning? Not sure.

Reply:
@tzdybal what do you prefer?

Reply:
Hmm good point.